internal/credential/vault/jobs_test.go

```diff
@@ -2,22 +2,152 @@ package vault
 
 import (
 	"context"
-	"fmt"
+	"path"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/hashicorp/boundary/internal/authtoken"
 	"github.com/hashicorp/boundary/internal/db"
 	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/host/static"
 	"github.com/hashicorp/boundary/internal/iam"
 	"github.com/hashicorp/boundary/internal/kms"
 	"github.com/hashicorp/boundary/internal/scheduler"
+	"github.com/hashicorp/boundary/internal/session"
+	"github.com/hashicorp/boundary/internal/target"
 	"github.com/hashicorp/go-hclog"
+	wrapping "github.com/hashicorp/go-kms-wrapping"
 	vault "github.com/hashicorp/vault/api"
+	"github.com/jinzhu/gorm"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+const testUpdateTokenStatusExpirationQuery = `
+update credential_vault_token
+   set status = ?,
+       last_renewal_time = now(),
+       expiration_time   = wt_add_seconds_to_now(?)
+ where token_hmac = ?;
+`
+
+func testVaultToken(t *testing.T,
+	conn *gorm.DB,
+	wrapper wrapping.Wrapper,
+	v *TestVaultServer,
+	cs *CredentialStore,
+	status TokenStatus,
+	expiration time.Duration) *Token {
+	t.Helper()
+	assert, require := assert.New(t), require.New(t)
+	kmsCache := kms.TestKms(t, conn, wrapper)
+	rw := db.New(conn)
+
+	secret, _ := v.CreateToken(t)
+	acc, err := secret.TokenAccessor()
+	require.NoError(err)
+	inToken, err := newToken(cs.PublicId, []byte(secret.Auth.ClientToken), []byte(acc), expiration)
```

Comment:
nit: This seems like it could be made a few lines shorter.  Since you are using secret.Auth to get ClientToken, consider doing the same to get `sec.Auth.Accessor` and bypass the require.NoError and the TokenAccessor() call.

Reply:
ty!